docs/sections/guides.md

```diff
@@ -147,3 +148,489 @@ run.
+# ...
+
+- id: "step_b"
+  uses: "./containers/step_B/
+```
+
+
+### Models and visualization
+
+Following the above, automate the other steps in your workflow using Popper. 
+This section shows examples for:
+- fitting a model to data 
+- generating model evaluation plots
+- using the model to make predictions on a hold-out dataset
+
+A first file, `src/models.py` defines the model this workflow uses:
+
+```python
+from sklearn import impute, preprocessing, compose, pipeline, linear_model, multioutput
+
+def _get_preprocessor(num_features , cat_features):
+
+    num_transformer = pipeline.Pipeline([
+        ("scale", preprocessing.StandardScaler()),
+        ("impute", impute.KNNImputer(n_neighbors = 10)),
+    ])
+
+    cat_transformer = pipeline.Pipeline([
+        ("impute", impute.SimpleImputer(strategy = "constant", fill_value = "missing")),
+        ("encode", preprocessing.OneHotEncoder(drop = "first")),
+    ] )
+
+    preprocessor = compose.ColumnTransformer(
+        [("num", num_transformer, num_features), 
+        ("cat", cat_transformer, cat_features)
+    ])
+    return preprocessor
+
+def get_lr_model(num_features, cat_features, C = 1.0):
+
+    model = pipeline.Pipeline([
+        ("pre", _get_preprocessor(num_features, cat_features)),
+        ("model", multioutput.MultiOutputClassifier(
+                    linear_model.LogisticRegression(penalty="l1", C = C, solver = "saga")
+        )),
+    ])
+    return model
+
+```
+
+A second script, `src/predict.py`, uses this model to generate the predictions
+on the hold-out dataset:
+
+ ```python
+import pandas as pd
+import os
+from models import get_lr_model
+
+DATA_PATH = "data/raw"
+PRED_PATH = "results/predictions"
+
+if __name__ == "__main__":
+
+    X_train = pd.read_csv(os.path.join(DATA_PATH, "training_set_features.csv")).drop(
+        "respondent_id", axis = 1
+    )
+
+    X_test = pd.read_csv(os.path.join(DATA_PATH, "test_set_features.csv")).drop(
+        "respondent_id", axis = 1
+    )
+
+    y_train = pd.read_csv(os.path.join(DATA_PATH, "training_set_labels.csv")).drop(
+        "respondent_id", axis = 1
+    )
+
+    sub = pd.read_csv(os.path.join(DATA_PATH, "submission_format.csv"))
+
+    num_features = X_train.columns[X_train.dtypes != "object"].values
+    cat_features = X_train.columns[X_train.dtypes == "object"].values
+
+    model = get_lr_model(num_features, cat_features, 1)
+    model.fit(X_train, y_train)
+    preds = model.predict_proba(X_test)
+
+    sub["h1n1_vaccine"] = preds[0][:, 1]
+    sub["seasonal_vaccine"] = preds[1][:, 1]
+    sub.to_csv(os.path.join(PRED_PATH, "baseline_pred.csv"), index = False)
+ ```
+
+Add this script as a step in the Popper workflow. This must come after the `get_data` 
+step
+
+```yaml
+- id: "predict"
+  uses: "./containers/"
+  args: ["python", "src/predict.py"]
+```
+
+Notes:
+- This use the same container as in the `notebook` step. Again, the final, 'canonical' analysis should be developed in the same environment
```

Comment:
this is so insightful. It is in general difficult to grasp unless one experiences it first-hand. It took me a couple of months to realize it, and I arrived to it thanks to following a container-native approach.